cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/ArtifactInspector.java

```diff
@@ -110,14 +110,15 @@
    * @param artifactFile the artifact file
    * @param parentClassLoader the parent classloader to use when inspecting plugins contained in the artifact.
    *                          For example, a ProgramClassLoader created from the artifact the input artifact extends
+   * @param additionalPlugins Additional plugin classes
    * @return metadata about the classes contained in the artifact
    * @throws IOException if there was an exception opening the jar file
    * @throws InvalidArtifactException if the artifact is invalid. For example, if the application main class is not
    *                                  actually an Application.
    */
   ArtifactClasses inspectArtifact(Id.Artifact artifactId, File artifactFile,
-                                  @Nullable ClassLoader parentClassLoader) throws IOException,
-                                                                                  InvalidArtifactException {
+                                  @Nullable ClassLoader parentClassLoader, @Nullable Set<PluginClass> additionalPlugins)
```

Comment:
in general, we should avoid allowing collections to be null. The caller should pass in an empty set if there are not additional plugins